0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+  "client_id": "3a2a1d9e-8640-4d2d-b06c-84f2cd613204",
+  "ledger": "http://usd-ledger.example",
+  "debit_account": "http://usd-ledger.example/accounts/alice",
+  "credit_account": "http://usd-ledger.example/accounts/bob",
+  "amount": "50",
+  "execution_condition": "cc:0:3:8ZdpKBDUV-KX_OnFZTsCWB_5mlCFI3DynX5f5H2dN-Y:2",
+  "expires_at": "2016-10-25T08:41:59.795Z"
+}
+```
+
+Response:
+
+```json
+HTTP/1.1 201 Created
+
+{
+  "id": "http://usd-ledger.example/transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204",
+  "client_id": "3a2a1d9e-8640-4d2d-b06c-84f2cd613204",
+  "ledger": "http://usd-ledger.example",
+  "debit_account": "http://usd-ledger.example/accounts/alice",
+  "credit_account": "http://usd-ledger.example/accounts/bob",
+  "amount": "50",
+  "execution_condition": "cc:0:3:8ZdpKBDUV-KX_OnFZTsCWB_5mlCFI3DynX5f5H2dN-Y:2",
+  "expires_at": "2016-10-25T08:41:59.795Z",
+  "state": "prepared",
+  "timeline": {
+    "prepared_at": "2015-06-16T00:00:01.000Z"
+  }
+}
+```
+
+#### Errors
+
+- [InsufficientFundsError][] - The `debit_account` would go below its `minimum_allowed_balance` if this transfer were executed
+- [UnprocessableEntityError][] - The request is formatted properly but contains an otherwise-unspecified semantic problem.
+- [AlreadyExistsError][] - The `client_id` supplied is already used by another transfer.
+- [InvalidUriParameterError][] - One of the [URL][] or [UUID][] parameters was not formatted properly.
+- [InvalidBodyError][] - The request body was not properly-formatted JSON, or did not match the Content-Type provided.
+- [UnsupportedCryptoConditionError][] - The transfer included an `execution_condition` whose feature bitstring requires functionality not implemented by this ledger.
+
+
+### Submit Fulfillment
+[Submit Fulfillment]: #submit-fulfillment
+
+Execute a transfer by submitting a [Crypto-Condition Fulfillment][]. To execute a transfer, the transfer MUST begin in the `prepared` state and the submitted fulfillment must satisfy the [Crypto-Condition][] in the transfer's `execution_condition` field. Doing so transitions the transaction to `executed`.
+
+**Authorization:** The owner of an account MUST be able to submit the fulfillment for a transfer that credits the account. A ledger MAY allow other account owners or unauthorized clients to submit a fulfillment. (If only the `credit_account` can submit a fulfillment, then crediting an account with a conditional transfer effectively requires permission from the account owner.)
+
+#### Request Format
+
+PUT to the `fulfillment` path of the transfer you want to execute, with the header `Content-Type: application/json` and a message body containing the fulfillment.
+
+**Metadata URL Name:** `transfer_fulfillment`. Suggested generic path: `/transfers/{client_id}/fulfillment`
+
+**Caution:** The ledger _MUST_ require the `Content-Type: application/json` header on this request. In the future, the API might also accept fulfillments in binary fulfillment at this same endpoint.
```

Comment:
The "Caution" callout here is to point out specific behavior that is innocuous now but would lead to spec incompatibility when we add a binary format.